File: Packages/HamStationKit/Sources/Models/RigState.swift
import Foundation

/// Current state of the connected rig, published via AsyncStream from RigConnection.
public struct RigState: Sendable, Equatable {
    /// Current VFO frequency in Hz.
    public var frequency: Double

    /// Current operating mode.
    public var mode: OperatingMode

    /// Whether PTT (push-to-talk) is currently active.
    public var pttActive: Bool

    /// Signal strength in dB (S-meter reading), if available.
    public var signalStrength: Int?

    public init(
        frequency: Double,
        mode: OperatingMode,
        pttActive: Bool,
        signalStrength: Int? = nil
    ) {
        self.frequency = frequency
        self.mode = mode
        self.pttActive = pttActive
        self.signalStrength = signalStrength
    }
}

File: Packages/shared/src/models/rig-state.ts
import type { OperatingMode } from './operating-mode';

export type ConnectionState = 'disconnected' | 'connecting' | 'connected' | 'reconnecting' | 'error';

export interface RigState {
  frequency: number;
  mode: OperatingMode;
  pttActive: boolean;
  signalStrength?: number;
}
